modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/configuration/CETConfigurationFactory.java

```diff
@@ -74,25 +108,112 @@ protected void modified(Map<String, Object> properties) throws Exception {
 				_cet = _cetManager.addCET(
 					cetConfiguration, companyId,
 					_getExternalReferenceCode(properties));
+
+				if (Objects.equals(
+						_cet.getType(),
+						ClientExtensionEntryConstants.TYPE_THEME_CSS)) {
+
+					_clientExtensionEntryRelLocalService.
+						deleteClientExtensionEntryRels(
+							companyId, _cet.getExternalReferenceCode());
+
+					_addThemeCSSClientExtensionEntryRel(companyId);
+				}
 			});
 	}
 
+	private void _addThemeCSSClientExtensionEntryRel(Long companyId)
+		throws PortalException {
+
+		ThemeCSSCET themeCSSCET = (ThemeCSSCET)_cet;
+
+		if (!Objects.equals(themeCSSCET.getScope(), "controlPanel")) {
+			return;
+		}
+
+		ClientExtensionEntryRel clientExtensionEntryRel =
+			_clientExtensionEntryRelLocalService.
+				fetchClientExtensionEntryRelByExternalReferenceCode(
+					themeCSSCET.getExternalReferenceCode(), companyId);
+
+		if (clientExtensionEntryRel != null) {
+			return;
+		}
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		User guestUser = company.getGuestUser();
+
+		Layout layout = _getControlPanelLayout(companyId);
+
+		_clientExtensionEntryRelLocalService.addClientExtensionEntryRel(
+			guestUser.getUserId(), layout.getGroupId(),
+			_portal.getClassNameId(Layout.class), layout.getPlid(),
+			_cet.getExternalReferenceCode(), _cet.getType(), StringPool.BLANK,
+			new ServiceContext());
+
+		int clientExtensionEntryRelsCount =
+			_clientExtensionEntryRelLocalService.
+				getClientExtensionEntryRelsCount(
+					_portal.getClassNameId(Layout.class), layout.getPlid(),
+					ClientExtensionEntryConstants.TYPE_THEME_CSS);
+
+		if (clientExtensionEntryRelsCount > 1) {
+			_log.error(
+				"Only one theme CSS client extension can be applied to the " +
+					"control panel at a time");
+		}
+	}
+
+	private Layout _getControlPanelLayout(long companyId)
+		throws PortalException {
+
+		Group group = _groupLocalService.fetchGroup(
+			companyId, GroupConstants.CONTROL_PANEL);
+
+		List<Layout> layouts = _layoutLocalService.getLayouts(
+			group.getGroupId(), true);
+
+		if (ListUtil.isEmpty(layouts)) {
+			throw new NoSuchLayoutException(
+				"Unable to get control panel layout");
+		}
+
+		return layouts.get(0);
```

Comment:
Is there only ever one control panel layout? If there is more than one, this logic seems like it wouldn't work.

Reply:
For instance, aren't there control panel pages on different sites. Those couldn't be the same layout could they?

Reply:
>For instance, aren't there control panel pages on different sites. Those couldn't be the same layout could they?

I tested it, the layout is the same for the entire instance, even though the site is different. For instance, if you deploy a brand new Liferay instance and open the browser's console and type `themeDisplay.getLayoutId()`, you'll get `1` for all control panel pages for all sites